src/Panel/DeprecatedPanel.php

```diff
@@ -0,0 +1,268 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace DebugKit\Panel;
+
+use Cake\Core\Plugin;
+use Cake\Event\Event;
+use Cake\Utility\Hash;
+use Composer\Json\JsonFile;
+use DebugKit\DebugPanel;
+
+/**
+ * Provides a list of deprecated methods for the current request
+ *
+ */
+class DeprecatedPanel extends DebugPanel
+{
+
+    /**
+     * The list of plugins within the application
+     *
+     * @var string[]
+     */
+    protected $_pluginPaths = [];
+
+    /**
+     * The list of Composer packages
+     *
+     * @var string[]
+     */
+    protected $_composerPaths = [];
+
+    /**
+     * File Types
+     *
+     * @var array
+     */
+    protected $_fileTypes = [
+        'Auth', 'Cache', 'Collection', 'Config', 'Configure', 'Console', 'Component', 'Controller',
+        'Behavior', 'Database', 'Datasource', 'Model', 'Template', 'View', 'Utility',
+        'Network', 'Routing', 'I18n', 'Log', 'Error', 'Event', 'Form', 'Filesystem',
+        'ORM', 'Filter', 'Validation'
+    ];
+
+    /**
+     * Get a list of plugins on construct for later use
+     */
+    public function __construct()
+    {
+        foreach (Plugin::loaded() as $plugin) {
+            $this->_pluginPaths[$plugin] = str_replace('/', DIRECTORY_SEPARATOR, Plugin::path($plugin));
+        }
+
+        $lockFile = new JsonFile(ROOT . DIRECTORY_SEPARATOR . 'composer.lock');
+        if ($lockFile->exists()) {
+            $lockContent = $lockFile->read();
+
+            $vendorDir = ROOT . DIRECTORY_SEPARATOR . 'vendor' . DIRECTORY_SEPARATOR;
+            $packages = array_merge($lockContent['packages'], $lockContent['packages-dev']);
+
+            foreach ($packages as $package) {
+                $this->_composerPaths[$package['name']] = $vendorDir . str_replace('/', DIRECTORY_SEPARATOR, $package['name']) . DIRECTORY_SEPARATOR;
+            }
+        }
```

Comment:
Should this be extracted into an object that can be used by both this and the include panel? Some sort of path classifier?